packages/console/src/components/RoleUsersTransfer/UserSourceBox.tsx

```diff
@@ -0,0 +1,111 @@
+import type { User } from '@logto/schemas';
+import { conditionalString } from '@silverhand/essentials';
+import type { ChangeEvent } from 'react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import Search from '@/assets/images/search.svg';
+import type { RequestError } from '@/hooks/use-api';
+import { onKeyDownHandler } from '@/utilities/a11y';
+
+import Checkbox from '../Checkbox';
+import Pagination from '../Pagination';
+import TextInput from '../TextInput';
+import UserAvatar from '../UserAvatar';
+import * as styles from './index.module.scss';
+
+type Props = {
+  onAddUser: (user: User) => void;
+  onRemoveUser: (user: User) => void;
+  selectedUsers: User[];
+};
+
+const pageSize = 20;
+
+const UserSourceBox = ({ selectedUsers, onAddUser, onRemoveUser }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [pageIndex, setPageIndex] = useState(1);
+  const [keyword, setKeyword] = useState('');
+
+  const { data } = useSWR<[User[], number], RequestError>(
+    `/api/users?page=${pageIndex}&page_size=${pageSize}&hideAdminUser=true${conditionalString(
+      keyword && `&search=${encodeURIComponent(`%${keyword}%`)}`
+    )}`
+  );
+
+  const [dataSource = [], totalCount] = data ?? [];
+
+  const handleSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setPageIndex(1);
+    setKeyword(event.target.value);
+  };
```

Comment:
Add debounce? From your video record, the list jumps on each typing. Should fire the API only once after user typing.

Reply:
Updated.